v7/static_tagcloud/static_tagcloud/__init__.py

```diff
@@ -0,0 +1,187 @@
+    # CSS border radius and the vertical margin. (The horizontal
+    # margin is set to zero.)
+    'round_factor': 0.0,
+}
+
+
+class StaticTagCloud(Task):
+    """Render tag clouds for various taxonomies."""
+
+    name = "render_static_tagcloud"
+
+    def _render_tagcloud_html(self, fn, tags, level_weights, config, lang, url_type):
+        """Create tag cloud HTML fragment."""
+        assert fn.startswith(self.site.config["OUTPUT_FOLDER"])
+        # Create fragment
+        html = engine.create_tag_cloud_html(config['name'], tags, level_weights)
+        # Determine location (for link rewriting)
+        url_part = fn[len(self.site.config["OUTPUT_FOLDER"]) + 1:]
+        src = os.path.normpath(os.sep + url_part)
+        src = "/".join(src.split(os.sep))
+        # Rewrite links
+        parser = lxml.html.HTMLParser(remove_blank_text=True)
+        doc = lxml.html.fragment_fromstring(html, parser)
+        self.site.rewrite_links(doc, src, lang, url_type)
+        html = (doc.text or '').encode('utf-8') + b''.join([lxml.html.tostring(child, encoding='utf-8', method='html') for child in doc.iterchildren()])
+        # Write result to disk
+        with open(fn, "wb") as html_file:
+            html_file.write(html)
+
+    def _render_tagcloud_css(self, css_fn, tags, level_weights, config):
+        """Create tag cloud CSS."""
+        assert css_fn.startswith(self.site.config["OUTPUT_FOLDER"])
+
+        css = engine.create_tag_cloud_css(config['name'], level_weights,
+                                          colors=config['colors'],
+                                          background_colors=config['background_colors'],
+                                          border_colors=config['border_colors'],
+                                          font_sizes=config['font_sizes'],
+                                          round_factor=config['round_factor'])
+        with open(css_fn, "wb") as css_file:
+            css_file.write(css.encode('utf-8'))
+
+    def _prepare_tagcloud(self, lang, config):
+        """Create tag cloud task."""
+        # Collect information
+        fn = os.path.join(self.site.config['OUTPUT_FOLDER'], config['filename'])
+        css_fn = os.path.join(self.site.config['OUTPUT_FOLDER'], config['style_filename'])
+        taxonomy_type = config['taxonomy_type']
+        posts_per_tag = self.site.posts_per_classification[taxonomy_type][lang]
+        taxonomy = self.site.taxonomy_plugins[taxonomy_type]
+
+        # Compose list of tags, their post count and links
+        def acceptor(post):
+            return True if self.site.config['SHOW_UNTRANSLATED_POSTS'] else post.is_translation_available(lang)
+
+        tag_count_url_list = []
+        for tag in natsort.humansorted(list(posts_per_tag.keys())):
+            tag_count_url_list.append((
+                taxonomy.get_classification_friendly_name(tag, lang),
+                len([post for post in posts_per_tag[tag] if acceptor(post)]),
```

Comment:
you can easily replace `acceptor` with the inlined logic, which can be simplified to
```py
self.site.config['SHOW_UNTRANSLATED_POSTS'] or post.is_translation_available(lang)
```